ssz/sedes/bitlist.py

```diff
@@ -0,0 +1,80 @@
+from typing import (
+    Sequence,
+    Union,
+)
+
+from ssz.exceptions import (
+    DeserializationError,
+    SerializationError,
+)
+from ssz.sedes.base import (
+    BaseCompositeSedes,
+)
+from ssz.utils import (
+    merkleize,
+    mix_in_length,
+    pack_bitvector_bitlist,
+)
+
+BytesOrByteArray = Union[bytes, bytearray]
+
+
+class Bitlist(BaseCompositeSedes[BytesOrByteArray, bytes]):
+    def __init__(self, bit_count: int) -> None:
+        if bit_count < 0:
+            raise TypeError("Max bit count cannot be negative")
+        self.bit_count = bit_count
+
+    #
+    # Size
+    #
+    is_fixed_sized = False
+
+    def get_fixed_size(self):
+        raise ValueError("byte list has no static size")
+
+    #
+    # Serialization
+    #
+    def serialize(self, value: BytesOrByteArray) -> bytes:
+        len_value = len(value)
+        if len_value > self.bit_count:
+            raise SerializationError(
+                f"Cannot serialize length {len_value} bit array as Bitlist[{self.bit_count}]"
+            )
+
+        if len_value == 0:
+            return b'\x01'
+
+        as_bytearray = [0] * (len_value // 8 + 1)
```

Comment:
is there a reason you aren't actually using a bytearray?  I think it will give you some added type/value safety since it won't allow assignment of values outside of the 0-256 range.

```python
as_bytearray = bytearray(len_value // 8 + 1)
```

Also, won't `len_value // 8 + 1` produce arrays with an extra element in the case that `len_value` is a clean multiple of `8`?